### DISCUSSION TITLE
transfer learning recommendations / best practices

### DISCUSSION THREAD

**Discussion by MarcMachaczek** [netket/netket#1497] [Q&A]
I want to train my VQS on my system for a range of external fields h (lets say parameters in general). In order to improve performance, I'd like to apply some kind of transfer learning to use the learned VQS for, say, h=0.1 as a starting point for the optimization for h=0.2. 
My approach is currently as follows: 
1) load the parameters from the previous trained model, add some gaussian noise, use them as new initial parameters
2) load the MC chains from the previous model (sampler_state) and use them as the new initial chains
3) using vqs.sample, let the MC chains run to adapt to the new initial noisy parameters
4) proceed with actual optimization for the new hamiltonian

I'm not 100% sure whether step 2) (and 3)) is necessary? Would this be the recommended way of doing it? Are there some best practices I'm missing? Also, are there any tips on how to choose the noise?

**Comment by gpescia** [comment]
Hi @MandMarc ,
I think your step 3) is not necessary at all, because there is actually a thermalization phase in netket, given by the vs.discard property. The larger this number is, the more time for the Markov Chain to adapt to the parameters.
Doing your step 2) might decrease the necessary size of this vs.discard number, because your chains are already more or less thermalized.
One question: What do you hope to obtain by adding some noise to your optimized parameters? It probably does not lead to worse results but I wonder if this really can help.

**Reply by MarcMachaczek** [reply]
Thanks for your input @gpescia! 
I observed that in a few cases where I initialized the VQS in the exact ground state (eg for the checkerboard fracton model), the chains starting from random configurations had problems converging to the ground state, while letting the model train from scratch allowed all chains to converge. This was my motivation for step 2). However, I observed (at least for my current project) that skipping step 2) leads to better results. My motivation for the noise is that, in particular close to phase transitions,  the wavefunction might change suddenly. In my intuition, adding small noise broadens the distribution allowing the optimization to more easily explore broader portions of hilbert space, which should help with not being stuck or concentrated in local minima close the previously optimized state. 
It happened (and still does sometimes) that initializing the the chains by random without any warm-up while transferring the previous optimized parameters without much/any noise leads to problems especially early during optimization, as the chains take very long to thermalize and gradient updates are therefore very noisy/incorrect.
In short, leaving out step 2) improved performance for me, but not adding noise/warmup leads to problems.
